src/flow/mod.rs

```diff
@@ -1191,14 +1191,28 @@ impl<'a> Migration<'a> {
         // etc.
         // println!("{}", mainline);
 
-        let mut uninformed_domain_nodes = domain_nodes.clone();
-        let deps = migrate::transactions::analyze_graph(
+        migrate::transactions::analyze_graph(
             &mainline.ingredients,
             mainline.source,
-            domain_nodes,
+            domain_new_nodes,
+            &mut mainline.deps,
         );
+
+        let mut uninformed_domain_nodes = mainline
+            .ingredients
+            .node_indices()
+            .filter(|&ni| ni != mainline.source)
+            .filter(|&ni| !mainline.ingredients[ni].is_dropped())
+            .map(|ni| {
+                (mainline.ingredients[ni].domain(), ni, new.contains(&ni))
+            })
+            .fold(HashMap::new(), |mut dns, (d, ni, new)| {
+                dns.entry(d).or_insert_with(Vec::new).push((ni, new));
+                dns
+        });
```

Comment:
This seems a little unfortunate. I wonder if we could rewrite this in terms of `new` too.

Reply:
This is used by `inform` which I think needs to inform all domains and do some computation over new nodes. So, it might be able to rewrite this, but the next bottle neck in the system is actually `materialization::commit`, so I don't think we should worry to much about it.

Reply:
Do you know what in `commit` is slow? I know it does a topological traversal of the entire graph, and that's bound to come back to bite us.

Reply:
As will the countless calls to `any_partial`.

Reply:
I haven't profiled it in depth. With 5k universes, a single call takes ~60ms. I don't think the traversal of the graph is that bad (if we only do it once), since `has_path` does `#bases` traversals and only takes ~20ms. Not sure about `any_partial` though.